asapdiscovery-genetics/asapdiscovery/genetics/calculate_rmsd.py

```diff
@@ -0,0 +1,151 @@
+from pathlib import Path
+
+import numpy as np
+from asapdiscovery.data.openeye import load_openeye_pdb, save_openeye_pdb
+from asapdiscovery.modeling.modeling import superpose_molecule
+
+
+def rmsd_alignment(
+    target_pdb: str,
+    ref_pdb: str,
+    final_pdb: str,
+    target_chain="A",
+    ref_chain="A",
+):
+    """Calculate RMSD of a molecule against a reference
+
+    Parameters
+    ----------
+    target_pdb : str
+        Path to PDB of protein to align.
+    ref_pdb : str
+        Path to PDB to align target against.
+    final_pdb : str
+        Path to save PDB of aligned target.
+    target_chain : str, optional
+        The chain of target which will be used for alignment, by default "A"
+    ref_chain : str, optional
+        The chain of reference which will be used for alignment, by default "A"
+
+    Returns
+    -------
+    float, str
+       RMSD after alignment, Path to saved PDB
+    """
+    protein = load_openeye_pdb(target_pdb)
+    ref_protein = load_openeye_pdb(ref_pdb)
+
+    aligned_protein, rmsd = superpose_molecule(
+        ref_protein, protein, ref_chain=ref_chain, mobile_chain=target_chain
+    )
+    pdb_aligned = save_openeye_pdb(aligned_protein, final_pdb)
+
+    return rmsd, pdb_aligned
+
+
+def select_best_colabfold(
+    results_dir: str,
+    seq_name: str,
+    pdb_ref: str,
+    chain="A",
+    final_pdb="aligned_protein.pdb",
+):
+    """Select the best seed output (repetition) from a ColabFold run based on its RMSD wrt the reference.
+
+    Parameters
+    ----------
+    results_dir : str
+        The directory containing the ColabFold results.
+    seq_name : str
+        The name we gave to the sequence in the csv file.
+    pdb_ref : str
+        The path to the PDB of te reference protein.
+    chain : str, optional
+        Chain of both reference and generated PDB that will be used, by default "A"
+    final_pdb : str, optional
+        Path to the PDB where aligned structure will be saved, by default "aligned_protein.pdb"
+
+    Returns
+    -------
+
+    float, str
+       RMSD after alignment, Path to saved PDB
+
+    Raises
+    ------
+    FileNotFoundError
+        The directory with ColabFold results does not exist
+    """
+
+    rmsds = []
+    seeds = []
+    file_seed = []
+
+    results_dir = Path(results_dir)
+    if not results_dir.exists():
+        raise FileNotFoundError(
+            f"A folder with ColbFold results {results_dir} does not exist"
+        )
+
+    default_CF = "*_unrelaxed_rank_001_alphafold2_ptm_model_1_seed_*.pdb"
```

Comment:
Make this a parameter that you can pass in with this default. Is there a way to make this simpler also? Or is this just what collabfold output looks like?

Reply:
That's the default name format for ColabFold, at least for the run in the tutorial. But I can pass it as a parameter in the rare case the name changes.